packages/jupyter/src/ConnectionStatusTray.tsx

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useState } from 'react';
+import { useThebeServer, useThebeSession } from 'thebe-react';
+import { useComputeOptions } from './providers';
+import type { ThebeEventData, ThebeEventType } from 'thebe-core';
+
+export function ConnectionStatusTray() {
+  const { thebe } = useComputeOptions();
+  const { connecting, ready: serverReady, error: serverError, events } = useThebeServer();
+  const { starting, ready: sessionReady, error: sessionError } = useThebeSession();
+  const [show, setShow] = useState(false);
+  const [unsub, setUnsub] = useState<() => void | undefined>();
+  const [status, setStatus] = useState<string>('[client] Connecting...');
+
+  const error = serverError || sessionError;
+  const ready = serverReady && sessionReady;
+  const busy = connecting || starting;
+
+  const handleStatus = (event: any, data: ThebeEventData) => {
+    setStatus(`[${data.subject}]: ${data.message}`);
+  };
+
+  useEffect(() => {
+    if (!events) return;
+    events.on('status' as ThebeEventType, handleStatus);
+  }, [events]);
+
+  useEffect(() => {
+    if (!thebe) return;
+    if (thebe?.useBinder || thebe?.useJupyterLite) {
+      if (busy || error) {
+        setShow(true);
+      } else if (ready) {
+        setTimeout(
```

Comment:
I think if we modify this in the future we should move to using radix Toast:
https://www.radix-ui.com/docs/primitives/components/toast

There is a bunch of accessibility built into that, as well as timeouts and stacking, and it can handle notifications from the full application not just here!